sky/provision/gcp/instance_utils.py

```diff
@@ -462,8 +464,9 @@ def call_operation(fn, timeout: int):
                 'message': f'Timeout waiting for operation {operation["name"]}',
                 'domain': 'wait_for_operation'
             }]
-            _log_errors(errors, None, zone)
+            msg = _format_message_from_errors(errors, None, zone)
             error = common.ProvisionerError('Operation timed out')
+            setattr(error, 'detailed_reason', msg)
```

Comment:
Should we add an attribute for the class and assign here, instead of `setattr`?

Reply:
This is only used for the usage collection, which uses `hasattr` to check `detailed_reason`. Maybe we can keep it this way?

Reply:
Oh ic. LGTM!